tensorly/regression/cp_plsr.py

```diff
@@ -86,6 +103,8 @@ def fit(self, X, Y):
         # Mean center the data, record info the object
         self.X_shape_ = T.shape(X)
         self.Y_shape_ = T.shape(Y)
+        original_X_ = T.copy(X)
+        original_Y_ = T.copy(Y)
```

Comment:
Can we get around the creation of those copy? Can we just work always on the centered versions of X and Y (we can uncenter if needed as we have the mean)?

Reply:
Hmm, I'm not sure how to avoid making a copy, either the original or the mean-centered version (what I changed into) of X and Y, if we still want to keep the $R^2$ calculation though. @aarmey do you have any idea?

Reply:
I think Jean's point is that right now you have two persistent copies—of both the original and mean-centered. Just keep a copy of one, and then build the other if/where you need it.

Reply:
Oh, then the latest change should make everything only mean-centered.